python/lsst/ap/association/l1dbproto.py

```diff
@@ -0,0 +1,296 @@
+ap_pipe/ap_verify.
+"""
+
+__all__ = ["AssociationL1DBProtoConfig",
+           "AssociationL1DBProtoTask"]
+
+import os
+
+from lsst.meas.algorithms.indexerRegistry import IndexerRegistry
+import lsst.afw.table as afwTable
+import lsst.afw.geom as afwGeom
+from lsst.ap.association import \
+    make_minimal_dia_object_schema, \
+    make_minimal_dia_source_schema
+from .afwUtils import \
+    convert_dia_source_to_asssoc_schema
+import lsst.l1dbproto.l1db as l1db
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+
+
+def _data_file_name(basename, module_name):
+    """Return path name of a data file.
+
+    Parameters
+    ----------
+    basename : `str`
+        Name of the file to add to the path string.
+    module_name : `str`
+        Name of lsst stack package environment variable.
+
+    Returns
+    -------
+    data_file_path : `str`
+       Fill path of the file to load from the "data" directory in a given
+       repository.
+    """
+    return os.path.join(os.environ.get(module_name), "data", basename)
+
+
+class AssociationL1DBProtoConfig(pexConfig.Config):
+    """Configuration parameters for the AssociationL1DBProtoTask.
+    """
+    indexer = IndexerRegistry.makeField(
+        doc='Select the spatial indexer to use within the database.',
+        default='HTM'
+    )
+    l1db_config = pexConfig.ConfigField(
+        dtype=l1db.L1dbConfig,
+        doc='Configuration for l1dbproto.',
+    )
+    filter_names = pexConfig.ListField(
+        dtype=str,
+        doc='List of filter names to store and expect from in this DB.',
+        default=['u', 'g', 'r', 'i', 'z', 'y'],
+    )
+
+    def setDefaults(self):
+        self.l1db_config.db_url = "sqlite:///l1dbproto.db"
+        self.l1db_config.isolation_level = "READ_UNCOMMITTED"
+        self.l1db_config.dia_object_index = "last_object_table"
+        self.l1db_config.read_sources_months = 12
+        self.l1db_config.read_forced_sources_months = 6
+        self.l1db_config.object_last_replace = True
+        self.l1db_config.explain = False
+        self.l1db_config.schema_file = _data_file_name(
+            "l1db-schema.yaml", "L1DBPROTO_DIR")
+        self.l1db_config.column_map = _data_file_name(
+            "l1db-ap-pipe-afw-map.yaml", "AP_ASSOCIATION_DIR")
+        self.l1db_config.extra_schema_file = _data_file_name(
+            "l1db-ap-pipe-schema-extra.yaml", "AP_ASSOCIATION_DIR")
+
+
+class AssociationL1DBProtoTask(pipeBase.Task):
+    """Task wrapping `lsst.l1dbproto` enabling it to be used in ap_association.
+
+    Handles computation of HTM indices, trimming of DIAObject catalogs to the
+    CCD geometry, and assuring input DIASource catalog schemas are compatible
+    with the db.
+    """
+
+    ConfigClass = AssociationL1DBProtoConfig
+    _DefaultName = "association_l1dbroto_task"
+
+    def __init__(self, **kwargs):
+
+        pipeBase.Task.__init__(self, **kwargs)
+        self.indexer = IndexerRegistry[self.config.indexer.name](
```

Comment:
So it turns out subtask doesn't work here because the function `makeHtmIndexer()` is what's registered and not the class HtmIndexer.   Looks like that was registered because 
`HtmIndexe`r takes depth in the constructor and not a config because it isn't a Task! https://github.com/lsst/meas_algorithms/blob/master/python/lsst/meas/algorithms/htmIndexer.py#L28 *To get this ticket merged I'm OK with leaving this as is.*  For the future though,  I found it confusing that it was a non-task masquerading as a subtask. @r-owenWhat do you think we should do going forward?

Reply:
Messed up the  @ mention ^^^ @r-owen

Reply:
From a *quick* glance at the `Indexer` classes I think they should be replaced by `sphgeom.Pixelization` classes, and we should have a registry of those instead.